<?xml version="1.0"?>
<launch>
  <!--<arg name="map_file" default="$(find erl_astar)/data/maps/sikang_map_3d_cell.yaml"/>-->
  <!--<arg name="map_file" default="$(find erl_astar)/data/maps/fla_warehouse1.yaml"/>-->
  <!--<arg name="map_file" default="$(find erl_astar)/data/maps/small_map_2d.yaml"/>-->
  <arg name="map_file" default="$(find erl_astar)/data/maps/large_map_3d.yaml"/>  


  <!-- Run the map server -->
  <node name="convert_legacy_gridmap" pkg="erl_map_ros" type="convert_legacy_gridmap" output="screen" args="$(arg map_file)">
    <param name="frame_id" value="map"/>
    <param name="data_type" value="uint16_t"/>
    <!--<param name="data_type" value="char"/>-->
  </node>
  
  <!-- Visualize in rviz -->
  <node pkg="erl_visualization" type="grid_msg_visualization" name="grid_visualization" output="screen">
    <param name="world_frame" value="map"/>
    <remap from="~grid_msg" to="map" />
  </node>
  
  <node pkg="rviz" type="rviz" name="rviz" output="screen"
    args="-d $(find erl_map_ros)/rviz/test_gridmap_server.rviz"/>  
</launch>

